libsolidity/ast/AST.cpp

```diff
@@ -319,6 +319,37 @@ FunctionDefinitionAnnotation& FunctionDefinition::annotation() const
 	return initAnnotation<FunctionDefinitionAnnotation>();
 }
 
+FunctionDefinition const& FunctionDefinition::resolveVirtual(
+	ContractDefinition const& _mostDerivedContract,
+	ContractDefinition const* _searchStart
+) const
+{
+	solAssert(!isConstructor(), "");
+	if (_searchStart == nullptr && !virtualSemantics())
```

Comment:
Why not `or`?

Reply:
`_searchStart == nullptr` means we are doing virtual lookup, not super lookup. Only in that case can we short-circuit. For super lookup, we have to go through the inheritance hierarchy even if the referenced function is not virtual, because we are searching upwards and not downwards.

Reply:
Ok!